web-client/src/views/FileDocument/StateDrivenFileInput.tsx

```diff
@@ -48,6 +57,53 @@ export const StateDrivenFileInput = connect<
 
     const fileOnForm = file || form[fileInputName] || form.existingFileName;
 
+    // Setting the filename here so that we can display it before validation
+    // finishes, otherwise a slow machine might have slight lag and allow the user
+    // to "choose file" again.
+    const [selectedFilename, setSelectedFilename] = useState('');
```

Comment:
See https://github.com/ustaxcourt/ef-cms/pull/5312/files#r1752579003 for why this is here. Happy to see alternatives.